theano/tensor/nnet/abstract_conv.py

```diff
@@ -1663,18 +1716,27 @@ def conv(self, img, kern, mode="valid", dilation=1, num_groups=1):
             raise ValueError(
                 'invalid dilation {}, expected {} values'.format(dilation,
                                                                  self.convdim))
+        if unshared and direction == "backprop weights":
+            if mode != "valid":
+                raise ValueError('conv mode for unshared backprop wrt weights must be "valid"')
+            # Do a transpose later to bring it to required shape
+            out_shape = (img.shape[0], kern.shape[0],
+                         kern.shape[2], kern.shape[3],
+                         img.shape[2] - kern.shape[2] + 1,
+                         img.shape[3] - kern.shape[3] + 1)
```

Comment:
The thing is, this is not the actual output shape of gradWeights. This is `(batch_size, num_filters, out_rows, out_cols, kH, kW)`, but the actual output shape is `(batch_size, out_rows, out_cols, num_filters, kH, kW)`. A transpose is done in the perform function to bring it into the required shape.
I could call `get_conv_gradweights_shape` and then shuffle around the values. Do you think I should do this?